nomad/csi_endpoint_test.go

```diff
@@ -473,40 +474,15 @@ func TestCSIPluginEndpoint_RegisterViaJob(t *testing.T) {
 
 	ns := structs.DefaultNamespace
 
-	job := mock.Job()
-	job.TaskGroups[0].Tasks[0].CSIPluginConfig = &structs.TaskCSIPluginConfig{
-		ID:       "foo",
-		Type:     structs.CSIPluginTypeMonolith,
-		MountDir: "non-empty",
-	}
+	deleteNodes := CreateTestPlugin(srv.fsm.State(), "foo")
```

Comment:
I'd probaly `defer deleteNodes()` here because it makes it clearer that it's a cleanup function